tests/integration-tests/tests/update/test_update.py

```diff
@@ -21,63 +22,106 @@
 from time_utils import minutes
 
 PClusterConfig = namedtuple(
-    "PClusterConfig", ["max_queue_size", "compute_instance", "s3_read_resource", "s3_read_write_resource"]
+    "PClusterConfig", [
+        "max_queue_size",
+        "compute_instance_type",
+        "compute_root_volume_size",
+        "s3_read_resource",
+        "s3_read_write_resource"
+    ]
 )
 
 
 @pytest.mark.dimensions("eu-west-1", "c5.xlarge", "alinux", "slurm")
 @pytest.mark.usefixtures("os", "scheduler")
-def test_update(instance, region, pcluster_config_reader, clusters_factory):
+def test_update(instance, region, pcluster_config_reader, clusters_factory, test_datadir):
     """
     Test 'pcluster update' command.
 
     Grouped all tests in a single function so that cluster can be reused for all of them.
     """
     s3_arn = "arn:aws:s3:::fake_bucket/*"
     init_config = PClusterConfig(
-        max_queue_size=5, compute_instance=instance, s3_read_resource=s3_arn, s3_read_write_resource=s3_arn
+        max_queue_size=5,
+        compute_instance_type=instance,
+        compute_root_volume_size=30,
+        s3_read_resource=s3_arn,
+        s3_read_write_resource=s3_arn
     )
-    cluster = _init_cluster(region, clusters_factory, pcluster_config_reader, init_config)
+    cluster = _init_cluster(clusters_factory, pcluster_config_reader, init_config)
+    command_executor = RemoteCommandExecutor(cluster)
+    slurm_commands = SlurmCommands(command_executor)
+
+    _verify_initialization(command_executor, slurm_commands, region, test_datadir, cluster, init_config)
 
     s3_arn_updated = "arn:aws:s3:::fake_bucket/fake_folder/*"
     updated_config = PClusterConfig(
         max_queue_size=10,
-        compute_instance="c4.xlarge",
+        compute_instance_type="c4.xlarge",
+        compute_root_volume_size=40,
         s3_read_resource=s3_arn_updated,
         s3_read_write_resource=s3_arn_updated,
     )
     _update_cluster(cluster, updated_config)
 
-    # test update
+    # verify updated parameters
     _test_max_queue(region, cluster.cfn_name, updated_config.max_queue_size)
-    _test_update_compute_instance_type(region, cluster, updated_config.compute_instance)
     _test_s3_read_resource(region, cluster, updated_config.s3_read_resource)
     _test_s3_read_write_resource(region, cluster, updated_config.s3_read_write_resource)
 
+    # verify updated params in OLD compute nodes
+    compute_nodes = slurm_commands.get_compute_nodes()
+    _test_compute_instance_type(
+        region, cluster.cfn_name, updated_config.compute_instance_type, compute_nodes[0], must_fail=True
```

Comment:
rather than having the must_fail logic you could assert that the instance type and the size still match the previous values.

Reply:
Good point